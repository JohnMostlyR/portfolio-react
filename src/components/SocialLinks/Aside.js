import styled from 'styled-components';

const Aside = styled.aside`
  float: right;
  padding: 0 1em 0 4em;
  background-image: linear-gradient(45deg, transparent 23%, #fff 23%);
  color: #575756;
`;

export default Aside;
