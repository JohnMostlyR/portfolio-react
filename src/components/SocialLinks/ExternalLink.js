import React from 'react';
import PropTypes from 'prop-types';
import FontAwesome from 'react-fontawesome';

import StyledA from './StyledA';
import VisuallyHiddenSpan from './VisuallyHiddenSpan'

const ExternalLink = (props) => (
    <StyledA href={props.href} target="_blank" rel="noopener noreferrer">
      <FontAwesome name={props.faIcon}/><VisuallyHiddenSpan>{props.description}</VisuallyHiddenSpan>
    </StyledA>
);

ExternalLink.propTypes = {
  description: PropTypes.string,
  href: PropTypes.string,
  faIcon: PropTypes.string,
};

ExternalLink.defaultProps = {
  faIcon: 'meh-o',
  description: 'ExternalLink to',
  href: '#',
};

export default ExternalLink;