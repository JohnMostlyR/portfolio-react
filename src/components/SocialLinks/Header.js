import styled from 'styled-components';
import {visuallyHidden} from '../../styles/mixins';

const Header = styled.h2`${visuallyHidden}`;

export default Header;