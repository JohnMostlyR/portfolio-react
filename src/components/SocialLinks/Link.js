import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import {visuallyHidden} from '../../styles/mixins';

const A = styled.a`
  color: #575756;
  text-decoration: none;
`;

const VisuallyHiddenSpan = styled.span`
  ${visuallyHidden}
`;

const Link = (props) => (
  <A href={props.href} target="_blank" rel="noopener noreferrer">
    <span className={props.iconClass} aria-hidden="true"/><VisuallyHiddenSpan>{props.description}</VisuallyHiddenSpan>
  </A>
);

Link.propTypes = {
  description: PropTypes.string,
  href: PropTypes.string,
  iconClass: PropTypes.string,
};

Link.defaultProps = {
  description: 'Link to',
  href: '#',
};

export default Link;