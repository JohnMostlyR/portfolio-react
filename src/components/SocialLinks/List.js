import styled from 'styled-components';

const List = styled.ul`
  display: flex;
  list-style: none;
`;

export default List;