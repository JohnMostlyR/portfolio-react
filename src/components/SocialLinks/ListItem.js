import styled from 'styled-components';
import typography from '../../styles/templates/typography';

const ListItem = styled.li`
  padding: .5em;
  transition: transform .2s;

  &:hover {
    transform: scale(1.5);
  }

  ${typography.doublePica}
  
  @media (min-width: 37.5em) and (max-height: 31.25em) {
    ${typography.pica}
  }
`;

export default ListItem;