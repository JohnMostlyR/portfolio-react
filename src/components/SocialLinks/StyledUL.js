import styled from 'styled-components';

const StyledUL = styled.ul`
  display: flex;
  list-style: none;
`;

export default StyledUL;
