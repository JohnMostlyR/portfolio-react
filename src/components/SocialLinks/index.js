import React, {Component} from 'react';
import ReactTooltip from 'react-tooltip';
import {FormattedMessage, injectIntl} from 'react-intl';

import ExternalLink from './ExternalLink';
import Aside from './Aside';
import Header from './Header';
import StyledUL from './StyledUL';
import StyledLI from './StyledLI';

class SocialLinks extends Component {
  playAnimation = (links) => {
    const INDEX = 0; // starting node index
    const START_DELAY = 3000; // milliseconds;
    const DELAY = 100; // milliseconds;

    setTimeout(function a(nodes, idx) {
      nodes[idx].style.transform = 'scale(1.5)';

      setTimeout(() => {
        nodes[idx].style.removeProperty('transform');
        if (idx < nodes.length - 1) {
          setTimeout(a, DELAY, nodes, idx += 1);
        } else {
          return true;
        }
      }, DELAY);
    }, START_DELAY, links, INDEX);
  };

  componentDidMount() {
    if (this.list.children.length) {
      this.playAnimation(this.list.children);
    }
  }

  render() {
    const externalLinks = [
      {
        icon: 'free-code-camp',
        name: 'freeCodeCamp',
        url: 'https://www.freecodecamp.org/mensae',
      },
      {
        icon: 'github',
        name: 'GitHub',
        url: 'https://github.com/Mensae',
      },
      {
        icon: 'linkedin-square',
        name: 'LinkedIn',
        url: 'https://nl.linkedin.com/in/jmeester',
      },
      {
        icon: 'codepen',
        name: 'Codepen',
        url: 'https://codepen.io/jmeester',
      },
    ];

    return (
        <Aside>
          <Header><FormattedMessage id='portfolio.social_links.header' defaultMessage='Also find me on:' /></Header>
          <StyledUL innerRef={e => this.list = e}>
            {
              externalLinks.map((externalLink, idx) => (
                  <StyledLI key={idx} data-tip={externalLink.name}>
                    <ExternalLink href={externalLink.url} faIcon={externalLink.icon} description={externalLink.name}/>
                  </StyledLI>
              ))
            }
          </StyledUL>
          <ReactTooltip />
        </Aside>
    );
  }
}

export default injectIntl(SocialLinks);
