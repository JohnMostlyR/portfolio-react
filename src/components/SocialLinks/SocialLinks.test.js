import React from 'react';
import ReactTooltip from 'react-tooltip';
import 'jest-styled-components';
import { shallow } from 'enzyme';
import { mountWithIntl } from '../../helpers/intl-enzyme-test-helper';

import SocialLinks from './index';
import Aside from './Aside';
import Header from './Header';
import StyledUL from './StyledUL';
import StyledLI from './StyledLI';

describe('SocialLinks', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mountWithIntl(
        <SocialLinks/>
    );

    // console.log(wrapper.debug());
  });

  it('should have an `Aside` component', () => {
    expect(wrapper.find(Aside)).toHaveLength(1);
  });

  it('should have an `Header` component', () => {
    expect(wrapper.find(Header)).toHaveLength(1);
  });

  it('should have an `StyledUL` component', () => {
    expect(wrapper.find(StyledUL)).toHaveLength(1);
  });

  it('should have four `StyledLI` components', () => {
    expect(wrapper.find(StyledLI)).toHaveLength(4);
  });

  it('each `StyledLI` component should have a data-tip property', () => {
    wrapper.find(StyledLI).forEach((node) => {
      expect(node.prop('data-tip')).toBeTruthy();
    })
  });

  it('should have a `ReactTooltip` component', () => {
    expect(wrapper.find(ReactTooltip)).toHaveLength(1);
  });

  it('Should render and match the snapshot', () => {
    const wrapper = shallow(<SocialLinks/>);
    expect(wrapper).toMatchSnapshot();
  });
});