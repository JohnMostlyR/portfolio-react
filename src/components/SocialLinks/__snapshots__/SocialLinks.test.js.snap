// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLinks Should render and match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SocialLinks />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "intl": Object {
        "formatMessage": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "intl": Object {
          "formatMessage": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
